lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -70,63 +69,57 @@ function generatePHDVersion(versionId) {
  * @param {Object} params - constructor params
  * @param {String} params.host - host for mongo
  * @param {String} params.port - port for mongo
- * @param {String} params.writeConcern - writeConcern for mongo
- * @param {String} params.replicaSet - replicaSet for mongo
- * @param {String} params.readPreference - readPreference for mongo
+ * @param {String} params.replicationGroupId - replication group id
+ * used here to generate version id's
+ * // Does backbeat use this at all? Can we make this optional and
+ * // set a default so when instantiate the client elsewhere don't need?
  * @param {String} params.database - name of database
  * @param {werelogs.Logger} params.logger - logger instance
  * @param {String} [params.path] - path for mongo volume
- * @param {String} [params.replicationGroupId] - replication group id
- * used here to generate version id's on puts. S3 instantiates the class
- * but backbeat does not
  */
 class MongoClientInterface {
-
     constructor(params) {
-        const { hosts,
-                writeConcern, replicaSet, readPreference,
-                path, database, logger,
-		replicationGroupId } = params;
-        const mongoUrl = format(
-            'mongodb://%s/?w=%s&replicaSet=%s&readPreference=%s',
-            hosts,
-            writeConcern,
-            replicaSet,
-            readPreference);
+        const { host, path, database, logger,
+            replicationGroupId, replicaSet } = params;
+        this.mongoUrl = format(
+            'mongodb://%s/?replicaSet=%s',
+            host,
+            replicaSet);
         this.logger = logger;
         this.client = null;
         this.db = null;
         this.path = path;
         this.replicationGroupId = replicationGroupId;
         this.database = database;
-        this.logger.info(`connecting to ${mongoUrl}`);
+        this.logger.debug(`connecting to ${this.mongoUrl}`);
```

Comment:
Please remove this log line as it's misleading. This is just a constructor, there is no connect method called.